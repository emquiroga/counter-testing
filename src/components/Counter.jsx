import { useCounter } from "../hooks/useCounter";
import ButtonCounter from "./ButtonCounter";

const Counter = () => {
  const { counter, handleChangeCounter, handleReset } = useCounter();
  return (
    <>
      <h2 role="counter">Counter: {counter} </h2>
      <ButtonCounter name="add" action={handleChangeCounter} />
      <button aria-label="reset" onClick={handleReset}>
        Reset
      </button>
      <button aria-label="rest" onClick={() => handleChangeCounter(-1)}>
        -1
      </button>
    </>
  );
};

export default Counter;
