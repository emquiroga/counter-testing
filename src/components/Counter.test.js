import { render, screen, fireEvent } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Counter from "./Counter"


describe("Testing component Counter", () => {
  
    beforeEach(()=> {
        render(<Counter />)
    })

    it("First snapshot", () => {
        expect(screen.getByRole("counter")).toMatchSnapshot();
    })

    it('Testing title and state rendering', () => {
        expect(screen.getByText("Counter: 0").tagName).toBe("H2");
    })
    it('Checking handleAdd function to work properly', () => {
        fireEvent.click(screen.getByLabelText("add"));
        expect(screen.getByRole("counter").textContent).toContain("Counter: 1");
    })
    it('Checking handleRest function to work properly', () => {
        userEvent.click(screen.getByLabelText("rest"));
        expect(screen.getByRole("counter").textContent).toContain("Counter: -1");
    })
    it('Simulating that someone is using the app', () => {
        const btnAdd = screen.getByLabelText("add");
        const btnRest = screen.getByLabelText("rest");

        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);

        expect(screen.getByRole("counter").textContent).toContain("Counter: 7");

        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);
        userEvent.click(btnRest);

        expect(screen.getByRole("counter").textContent).toContain("Counter: -4");

    })
    it('Testing reset button', () => {
        const btnAdd = screen.getByLabelText("add");

        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);
        userEvent.click(btnAdd);

        userEvent.click(screen.getByLabelText("reset"));

        expect(screen.getByRole("counter").textContent).toContain("Counter: 0")

    })
})
