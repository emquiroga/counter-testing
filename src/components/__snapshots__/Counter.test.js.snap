// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component Counter First snapshot 1`] = `
<h2
  role="counter"
>
  Counter: 
  0
   
</h2>
`;
